beta/plugins/remark-header-custom-ids.js

```diff
@@ -31,33 +31,51 @@ module.exports = ({
   return function transformer(tree) {
     visit(tree, 'heading', (node) => {
       const children = node.children;
+      const rawHeader = toString(node);
+      const match = /^.+(\s*\{.*\}\s*)$/u.exec(rawHeader);
+      /**
+       * we check if there is a heading and custom id with regex.
+       * If there is a match then we set the id to be the match[1] which is the custom id
+       * else we have to generate id
+       */
+      let id =
+        match && match.length === 2
+          ? match[1]
+          : slugs.slug(rawHeader, maintainCase);
+
       let tail = children[children.length - 1];
 
-      // A bit weird: this is to support MDX 2 comments in expressions,
-      // while we’re still on MDX 1, which doesn’t support them.
-      if (!tail || tail.type !== 'text' || tail.value !== '/}') {
-        return;
-      }
+      /**
+       * If there was a match then we might want to return when they dont have proper heading ids
+       * Else we generate ids
+       */
+      if (match) {
+        if (!tail || tail.type !== 'text' || tail.value !== '/}') {
+          return;
```

Comment:
These early exits didn't seem right. We want to assign ID from a slug regardless of whether we succeeded to modify the tail.